packages/itwin/grouping-mapping-widget/src/widget/components/SelectIModel.tsx

```diff
@@ -26,20 +27,29 @@ const SelectIModel = ({
   onCancel,
   backFn,
 }: SelectIModelProps) => {
-  const apiContext = useContext(ApiContext);
+  const { getAccessToken, prefix } = useGroupingMappingApiConfig();
+  const [accessToken, setAccessToken] = useState<AccessToken>();
 
   const apiOverrides = useMemo<ApiOverrides<IModelFull[]>>(
-    () => ({ serverEnvironmentPrefix: apiContext.prefix }),
-    [apiContext.prefix]
+    () => ({ serverEnvironmentPrefix: prefix }),
+    [prefix]
   );
 
+  useEffect(() => {
+    const fetchAccessToken = async () => {
+      const accessToken = await getAccessToken();
+      setAccessToken(accessToken);
+    };
+    void fetchAccessToken();
+  }, [getAccessToken]);
+
   return (
     <div className='imodel-grid-container'>
       <div className='imodel-grid'>
         <IModelGrid
           projectId={projectId}
           onThumbnailClick={onSelect}
-          accessToken={apiContext.accessToken}
+          accessToken={accessToken}
```

Comment:
There is a chance that an expired `accessToken` will be passed into `IModelGrid`. New `accessToken` will only be fetched if `getAccessToken` function changes, not when the token expires. It would be best if the `IModelGrid` component would accept a function to get the token instead of the token itself. Chance of the "expired" `accessToken` issue occuring is relatively low, so we can leave this one for another PR.